File: corescript-vscode/src/aiInstructions.ts
/**
 * AI instructions for generating Paracore gallery scripts.
 *
 * CANONICAL SOURCE: ..\\docs\\copilot-instructions.md (shared Paracore docs).
 *
 * These instructions are written to .github/copilot-instructions.md
 * when a workspace is scaffolded, teaching VS Code AI assistants
 * (Copilot, Cline, etc.) how to generate Paracore gallery scripts
 * with editable Params classes.
 */

import * as fs from "fs";
import * as path from "path";

function loadInstructions(): string {
  // Search paths for the canonical file — try shared docs first
  const searchPaths = [
    // Dev: shared container docs (__dirname = src/ → corescript-vscode/ → paracore/ → container/)
    path.join(__dirname, "..", "..", "..", "docs", "copilot-instructions.md"),
    // Production: bundled in extension root (__dirname = dist/)
    path.join(__dirname, "..", "copilot-instructions.md"),
  ];

  for (const p of searchPaths) {
    if (fs.existsSync(p)) {
      return fs.readFileSync(p, "utf-8");
    }
  }

  console.warn(
    "Paracore: copilot-instructions.md not found. AI script generation may produce suboptimal code."
  );
  return `# Script Context: Paracore Tool Project
# All logic goes into the Scripts/ folder.

Generate C# Revit API scripts for the Paracore runtime (CoreScript.Engine).

See the Paracore documentation for complete scripting reference.
`;
}

export const COPILOT_INSTRUCTIONS: string = loadInstructions();

File: rap-server/server/api/ai_instructions.py
"""
Canonical AI instructions for generating Paracore gallery scripts.
SOURCE OF TRUTH: copilot-instructions.md in the paracore repo root.
This module reads from the canonical file at import time (dev) or falls
back to an inline default (production bundles).
"""
import os

_instructions_path = os.path.join(os.path.dirname(__file__), "..", "..", "..", "copilot-instructions.md")
_instructions_path = os.path.normpath(_instructions_path)

try:
    with open(_instructions_path, "r", encoding="utf-8") as f:
        COPILOT_INSTRUCTIONS = f.read()
except FileNotFoundError:
    # Fallback for production deployments where the md file isn't present
    COPILOT_INSTRUCTIONS = """# Script Context: Paracore Tool Project
# All logic goes into the Scripts/ folder.
# ...

Generate C# Revit API scripts for the Paracore runtime (CoreScript.Engine).
See docs/copilot-instructions.md for the full canonical content.
"""
